smart_open/hdfs.py

```diff
@@ -18,8 +18,35 @@
 import logging
 import subprocess
 
+from six.moves.urllib import parse as urlparse
```

Comment:
offtopic: any plans for dropping py2 support in `smart_open`?

Reply:
Yes, let's do it in a separate PR, as soon as possible. The rationale is the same. People who cannot upgrade to Py3 can continue to use existing versions of smart_open.

CC @piskvorky

Reply:
Yes, let's. With good communication it shouldn't be an issue.

Are there any stats on smart_open vs Python version?

Reply:
Should be a way to do it via https://packaging.python.org/guides/analyzing-pypi-package-downloads/, I'll have a look at it during the week.